app/services/PspDetailsService.scala

```diff
@@ -18,86 +18,106 @@ package services
 
 import com.google.inject.Inject
 import config.FrontendAppConfig
+import connectors.MinimalConnector
 import connectors.SubscriptionConnector
 import connectors.cache.UserAnswersCacheConnector
 import controllers.amend.routes
 import controllers.company.{routes => compRoutes}
 import controllers.individual.{routes => indRoutes}
 import controllers.partnership.{routes => partRoutes}
 import models.SubscriptionType.Variation
-import models.register.RegistrationCustomerType.{NonUK, UK}
+import models.register.RegistrationCustomerType.{UK, NonUK}
 import models.register.RegistrationLegalStatus._
-import models.register.{RegistrationDetails, RegistrationIdType}
+import models.register.{RegistrationIdType, RegistrationDetails}
 import models.{CheckMode, UserAnswers}
-import pages.company.{CompanyAddressPage, CompanyEmailPage, CompanyPhonePage}
+import pages.company.{CompanyPhonePage, CompanyAddressPage, CompanyEmailPage}
 import pages.individual._
 import pages.partnership._
 import pages.register._
-import pages.{PspIdPage, RegistrationDetailsPage, SubscriptionTypePage, company => comp}
+import pages.{SubscriptionTypePage, PspIdPage, RegistrationDetailsPage, company => comp}
 import play.api.i18n.Messages
-import play.api.libs.json.{JsObject, JsValue, Json}
+import play.api.libs.json.{Json, JsObject, JsValue}
 import play.api.mvc.Call
 import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.viewmodels.SummaryList.{Action, Key, Row, Value}
+import uk.gov.hmrc.viewmodels.SummaryList.{Key, Value, Row, Action}
 import uk.gov.hmrc.viewmodels.Text.Literal
 import uk.gov.hmrc.viewmodels._
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{Future, ExecutionContext}
 import scala.util.Try
 
 class PspDetailsService @Inject()(appConfig: FrontendAppConfig,
                                   subscriptionConnector: SubscriptionConnector,
-                                  userAnswersCacheConnector: UserAnswersCacheConnector) extends CYAService {
+                                  userAnswersCacheConnector: UserAnswersCacheConnector,
+                                  minimalConnector: MinimalConnector) extends CYAService {
 
   val halfWidth: Seq[String] = Seq("govuk-!-width-one-half")
   val thirdWidth: Seq[String] = Seq("govuk-!-width-one-third")
   def nextPage: String = routes.DeclarationController.onPageLoad().url
 
+  private def returnUrlAndLink(regInfo:RegistrationDetails, ua:UserAnswers, rlsFlag:Boolean)(implicit messages: Messages,
+    hc: HeaderCarrier, ec: ExecutionContext):JsObject = {
+    if (rlsFlag) {
+      Json.obj()
+    } else {
+      val returnUrl = Json.obj(
+        "returnUrl" -> appConfig.returnToPspDashboardUrl
+      )
+      val returnLink = Json.obj(
+        regInfo.legalStatus match {
```

Comment:
The repetition of this match case in the same file can be avoided if this method accepts the Individual/Company name instead of RegDetails